google/cloud/aiplatform/metadata/metadata_store.py

```diff
@@ -0,0 +1,140 @@
+        )
+        self._gca_resource = self._get_gca_resource(
+            resource_name=metadata_store_name, allow_str_id=True
+        )
+
+    @classmethod
+    def create(
+        cls,
+        metadata_store_id: str = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+        encryption_spec_key_name: Optional[str] = None,
+    ) -> "MetadataStore":
+        """Creates a new MetadataStore if it does not exist.
+
+        Args:
+            metadata_store_id (str):
+                The {metadatastore} portion of the resource name with
+                the format:
+                projects/{project}/locations/{location}/metadataStores/{metadatastore}
+                If not provided, the MetadataStore's ID will be set to "default" to create a default MetadataStore.
+            project (str):
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+            encryption_spec_key_name (Optional[str]):
+                Optional. The Cloud KMS resource identifier of the customer
+                managed encryption key used to protect the metadata store. Has the
+                form:
+                ``projects/my-project/locations/my-region/keyRings/my-kr/cryptoKeys/my-key``.
+                The key needs to be in the same region as where the compute
+                resource is created.
+
+                If set, this MetadataStore and all sub-resources of this MetadataStore will be secured by this key.
+
+                Overrides encryption_spec_key_name set in aiplatform.init.
+
+
+        Returns:
+            metadata_store (MetadataStore):
+                Instantiated representation of the managed metadata store resource.
+
+        """
+        api_client = cls._instantiate_client(location=location, credentials=credentials)
+        gapic_metadata_store = gca_metadata_store.MetadataStore(
+            encryption_spec=initializer.global_config.get_encryption_spec(
+                encryption_spec_key_name=encryption_spec_key_name
+            )
+        )
+
+        try:
+            api_client.create_metadata_store(
+                parent=initializer.global_config.common_location_path(
+                    project=project, location=location
+                ),
+                metadata_store=gapic_metadata_store,
+                metadata_store_id=metadata_store_id,
+            )
+        except exceptions.AlreadyExists:
+            logging.info("MetadataStore %s already exists" % metadata_store_id)
```

Comment:
@sasha-gitg when executing this in the notebook, this is not catched, and I see the stacktrace in the notebook. Is there something that I need to configure extra to "catch" exceptions? Or am I doing it wrong? I saw @jialuzh had the same and he is able to see the log: https://github.com/jialuzh/python-aiplatform/blob/dev-test/google/cloud/aiplatform/metadata/metadata_store.py#L167

Reply:
Please provide the exception output. It's most likely a different exception we need to catch.

Reply:
nvm was missing to call `.result()` in the above call... By doing so, the exception is correct